hs_core/templates/search/search.html

```diff
@@ -50,190 +36,515 @@ <h4 class="panel-title">
+                    event.preventDefault();
+                    localStorage.clear();
+                    $("#search-field").submit();
+                }
+            });
+
+            $(".faceted-selections").click(function () {
+                var tokens = $(this).val().split(",");
+                var key = tokens[0];
+                var value = tokens[1];
+                var textSearch = $("#id_q").val();
+                var searchURL = "?q=" + textSearch;
+                var windowPath = window.location.pathname;
+                var requestURL =  windowPath + searchURL;
+
+                if($(this).is(":checked")) {
+                    requestURL = requestURL + "&selected_facets="+key+"_exact:"+value
+
+                    $(".faceted-selections").each(function () {
+                        var val = localStorage[$(this).attr("id")];
+                        var isChecked = val === 'true';
+                        if(isChecked){
+                            var arr = $(this).val().split(",");
+                            var key = arr[0];
+                            var value = arr[1];
+                            requestURL += "&selected_facets="+key+"_exact:"+value;
+                        }
+                    });
+                    window.location = requestURL;
+                    localStorage[$(this).attr("id")] = this.checked;
+                }else{
+                    localStorage.removeItem($(this).attr("id"));
+                    var textSearch = $("#id_q").val();
+                    var updateURL = "/search/?q=" + textSearch;
+                    $(".faceted-selections").each(function () {
+                        var val = localStorage[$(this).attr("id")];
+                        var isChecked = val === 'true';
+                        if(isChecked){
+                            var arr = $(this).val().split(",");
+                            var key = arr[0];
+                            var value = arr[1];
+                            updateURL += "&selected_facets="+key+"_exact:"+value;
+                        }
+                    });
+                    window.location = updateURL;
+                }
+            });
+
             $(".faceted-selections").each(function () {
                 var val = localStorage[$(this).attr("id")];
-                var isChecked = val !== undefined ? val == "true" : false;
+                var isChecked = val !== undefined ? val == 'true' : false;
                 $(this).prop("checked", isChecked);
             });
-        $("#faceted-search").click(function(){
-            localStorage.clear();
+
+
+            $('[data-toggle="popover"]').popover({
+                html: true,
+                container: '#items',
+                content: '<p>This search box supports <a href="https://cwiki.apache.org/confluence/display/solr/Searching" target="_blank">SOLR Query syntax</a>.<br />' +
+                '"*" means show all results </p>',
```

Comment:
@xuezhaokun @ironfroggy I suggest deleting the * documentation. But as to danger, I don't consider exposing the SOLR interface -- which haystack is designed to do -- to be dangerous. 

So: 
- yes to confusing
- yes to delete '*' comment. 
- no to dangerous.